### PR TITLE
PSR-1100 addressed basic details issues

### PR DESCRIPTION
**Pull request by westwater**

### CODE REVIEW

File: app/pages/nonsipp/BasicDetailsCompletedPage.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package pages.nonsipp
+
+import pages.QuestionPage
+import models.SchemeId.Srn
+import play.api.libs.json.{__, JsPath}
+import viewmodels.models.SectionCompleted
+
+case class BasicDetailsCompletedPage(srn: Srn) extends QuestionPage[SectionCompleted] {
```
Comment (kberec) [inline app/pages/nonsipp/BasicDetailsCompletedPage.scala:24]: Do we need to set this value up in FE mongo cache on GET from ETMP?

File: app/controllers/nonsipp/CheckReturnDatesController.scala [hunk 2]
```diff
@@ -77,11 +79,40 @@ class CheckReturnDatesController @Inject()(
...
           .fold(
             formWithErrors => Future.successful(BadRequest(view(formWithErrors, viewModel))),
-            value =>
-              for {
-                updatedAnswers <- Future.fromTry(request.userAnswers.set(CheckReturnDatesPage(srn), value))
-                _ <- saveService.save(updatedAnswers)
-              } yield Redirect(navigator.nextPage(CheckReturnDatesPage(srn), mode, updatedAnswers))
+            value => {
+
+              val updatedUserAnswers = request.userAnswers.get(CheckReturnDatesPage(srn)) match {
```
Comment (kberec) [inline app/controllers/nonsipp/CheckReturnDatesController.scala:84]: Quite a bit of logic added here... 
- would it be useful it this is moved to a separate method?
- do we have tests for this logic?

File: app/pages/nonsipp/BasicDetailsCompletedPage.scala (hunk 1 above)
Comment (westwater) [reply app/pages/nonsipp/BasicDetailsCompletedPage.scala:24]: No, since we don't go back to amend basic details after a submission theres no need

**Review by kberec — DISMISSED** [review]

**Review by kberec — APPROVED** [review]

**Review by kberec — APPROVED** [review]

File: app/controllers/nonsipp/CheckReturnDatesController.scala (hunk 2 above)
Comment (westwater) [reply app/controllers/nonsipp/CheckReturnDatesController.scala]: I've shrunk the logic quite a bit here now and added unit tests

**Pull request merged**